### ISSUE TITLE
[v2] Custom NodeInterface component can't inject Temporary Connection Handler

### ISSUE THREAD

**Issue by luhof** [newcat/baklavajs#408]
Hi newcat! Congrats for the work on the v2 💪!

I've been trying to fiddle with custom components to work around some features, like customizing the Node component to make it fully draggable and not just with the handler etc. It works well!

However, I wanted to use a custom `NodeInterface` component. My main objective was to play with the draggable handle to achieve some "magnetism".

I copy-pasted the [default NodeInterface component](https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/src/node/NodeInterface.vue) as linked from the [documentation](https://v2.baklava.tech/visual-editor/customization.html#custom-components).
However, some internal imports used by the NodeInterface component are not available when using the component in my project, mostly imports from `../editor/temporaryConnection.ts`.
https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/src/node/NodeInterface.vue#L34-L37

For now I just tried to empirically re-use some bits of the temporary connection to my project without any success, and my interfaces are simply not draggable.

Is there any reason not to export the `useTemporaryConnection` composable?
https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/src/editor/temporaryConnection.ts#L12

If so, is there any alternative you'd suggest to customize the `NodeInterface` component?

I hope I'm clear enough, have a nice day!

**Comment by Doltario** [comment]
Hey hi guys :)

I couldn't agree more with @luhof : Congrats for the work @newCat !

However I was wondering something : I am trying to customize the connections between my Nodes, and I think I am facing an issue similar to Luhof’s one.

I used `https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/src/connection/ConnectionWrapper.vue`.

But I can’t find exports for both following lines:
 ```javascript
import resolveDom, { IResolvedDomElements } from "./domResolver";
import { TemporaryConnectionState } from "./connection";
```

I think It is related to what Luhof pointed, but do not hesitate to let me know if I did post in the wrong place.

Have a nice day guys !

**Comment by Farnsi** [comment]
Same problem v2.4.5.
Does the dist use other / older sources? 
[useTemporaryConnection](https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/src/editor/index.ts) is exported, but could not be imported.

**Comment by Doltario** [comment]
Hi @newcat, any update on that ? I can't really delay the migration to BaklavaV2 anymore 😅

**Comment by newcat** [comment]
Done in [v2.5.0](https://github.com/newcat/baklavajs/releases/tag/v2.5.0)

**Comment by Doltario** [comment]
Hi again, if I understood well the commit, the commit https://github.com/newcat/baklavajs/commit/43e1de0842c493ffa0c98f3114755a40da592476 fixes the original issue. 

For this problematic (https://github.com/newcat/baklavajs/issues/408#issuecomment-2149358087), should I create a new issue, or am I misusing the customization feature ?

**Comment by newcat** [comment]
@Doltario Should be fixed in [v2.5.1](https://github.com/newcat/baklavajs/releases/tag/v2.5.1)
